File: algorithms/1_easy/P/02614_prime_in_diagonal/solution.py
class Solution:
    def is_prime(self, num):
        if num == 2:
            return True
    
        if num <= 1 or num % 2 == 0:
            return False

        sq = int(sqrt(num))
        for i in range(3, sq + 1, 2):
            if num % i == 0:
                return False

        return True

    def diagonalPrime(self, nums: List[List[int]]) -> int:
        n = len(nums)
        max_num = 0

        for i in range(n):
            if max_num < nums[i][i] and self.is_prime(nums[i][i]):
                max_num = nums[i][i]

            if max_num < nums[i][n - 1 - i] and self.is_prime(nums[i][n - 1 - i]):
                max_num = nums[i][n - 1 - i]

        return max_num

File: algorithms/1_easy/P/02614_prime_in_diagonal/solution.rs
impl Solution {
    fn is_prime(num: i32) -> bool {
         if num == 2 {
            return true;
        }

        if num <= 1 || num % 2 == 0 {
            return false;
        }

        let sq = (num as f64).sqrt() as i32;
        for i in (3..=sq).step_by(2) {
            if num % i == 0 {
                return false;
            }
        }

        true
    }

    pub fn diagonal_prime(nums: Vec<Vec<i32>>) -> i32 {
        let n = nums.len();
        let mut max = 0_i32;

        for i in 0..n {

            if max < nums[i][i] && Self::is_prime(nums[i][i]) {
                max = nums[i][i];
            }

            if max < nums[i][n - 1 - i] && Self::is_prime(nums[i][n - 1 - i]) {
                max = nums[i][n - 1 - i];
            }
        }

        max
    }
}
